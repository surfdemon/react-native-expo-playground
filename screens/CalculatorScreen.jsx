import React, {useState} from 'react';
import {StyleSheet, ScrollView, View, Text, TouchableOpacity} from 'react-native';
import CalculatorButton from '../components/CalculatorButton';

const CalculatorScreen = ({navigation}) => { 
    //Setup useStates.
    const [calculatorState, setCalculatorState] = useState({
        currentValue: '0', 
        operatorSymbol: null,
        previousValue: null,
    });
    React.useEffect(() => {    
    });

    const operatorPress = (sym) => {
        const { currentValue, previousValue, operatorSymbol } = calculatorState;
        setCalculatorState({ previousValue: currentValue, currentValue: '0', operatorSymbol: sym });
    }

    const calculatePress = () => {
        const { currentValue, previousValue, operatorSymbol } = calculatorState;
        let newValue; 
        if (operatorSymbol === '+'){
            newValue = `${parseFloat(previousValue) + parseFloat(currentValue)}`.toString();
            setCalculatorState({ currentValue: newValue, operatorSymbol: null, previousValue: null});
        } else if (operatorSymbol === '-'){
            newValue = `${previousValue - currentValue}`;
            setCalculatorState({ currentValue: newValue, operatorSymbol: null, previousValue: null});
        } else if (operatorSymbol === '/'){
            newValue = `${previousValue / currentValue}`;
            setCalculatorState({ currentValue: newValue, operatorSymbol: null, previousValue: null});
        } else if (operatorSymbol === '*'){
            newValue = `${previousValue * currentValue}`;
            setCalculatorState({ currentValue: newValue, operatorSymbol: null, previousValue: null});
        }
    }

    const clearSums = () => {
       setCalculatorState({ currentValue: '0', operatorSymbol: null, previousValue: null});
    }

    const pointPress = (num) => {
        const { currentValue, previousValue, operatorSymbol } = calculatorState;
        setCalculatorState({ currentValue : currentValue + num, previousValue: previousValue, operatorSymbol: operatorSymbol});
    }

    const numPress = (num) => {
        const { currentValue, previousValue, operatorSymbol } = calculatorState;

        if (num === '.' && !currentValue.includes('.')) {
            //if num === . and currentValue does not include a . then add it to the end.
            return setCalculatorState({ currentValue: currentValue + num, previousValue: previousValue, operatorSymbol: operatorSymbol});
        } else if ( num !== '.') {
            //if num is not a . carry on 
            if (currentValue === '0'){
                //if currentValue is a 0 then replace it with new num
                setCalculatorState({  currentValue: num, previousValue: previousValue, operatorSymbol: operatorSymbol  });   
            } else {
                // if currentValue is not 0 then add new num to the end of currentValue
                setCalculatorState({ currentValue: currentValue + num, previousValue: previousValue, operatorSymbol: operatorSymbol});
            }
        }
    }

    return (
        <View style={[styles.mainRow]}>
            <View style={[styles.container]}>
                <View style={[styles.sumArea]}>
                    <Text id='sum' testID='sum' style={[styles.text, styles.sum]}>{calculatorState.currentValue}</Text>
                </View>
                <View style={[styles.row]}>
                    <CalculatorButton title="C" onPress={() => clearSums('C')} isTripleWidthButton testID='C' />
                    <CalculatorButton title="*" onPress={() => operatorPress('*')} symbol={calculatorState.operatorSymbol} isTripleWidthButton testID='*' />
                </View>    
                <View style={[styles.row]}>
                    <CalculatorButton title="1" onPress={() => numPress('1')} testID='1' />
                    <CalculatorButton title="2" onPress={() => numPress('2')} testID='2' />
                    <CalculatorButton title="3" onPress={() => numPress('3')} testID='3' />
                    <CalculatorButton title="+" onPress={() => operatorPress('+')} symbol={calculatorState.operatorSymbol} testID='+' />
                </View>    
                <View style={[styles.row]}>
                    <CalculatorButton title="4" onPress={() => numPress('4')} testID='4' />
                    <CalculatorButton title="5" onPress={() => numPress('5')} testID='5' />
                    <CalculatorButton title="6" onPress={() => numPress('6')} testID='6' />
                    <CalculatorButton title="-" onPress={() => operatorPress('-')} symbol={calculatorState.operatorSymbol} testID='-' />
                </View>
                <View style={[styles.row]}>
                    <CalculatorButton title="7" onPress={() => numPress('7')} testID='7' />
                    <CalculatorButton title="8" onPress={() => numPress('8')} testID='8' />
                    <CalculatorButton title="9" onPress={() => numPress('9')} testID='9' />
                    <CalculatorButton title="/" onPress={() => operatorPress('/')} symbol={calculatorState.operatorSymbol} testID='/' />
                </View> 
                <View style={[styles.row]}>
                    <CalculatorButton title="0" onPress={() => numPress('0')} isDoubleWidthButton testID='0' />
                    <CalculatorButton title="." onPress={() => numPress('.')} testID='dot' />
                    <CalculatorButton title="=" onPress={() => calculatePress('=')} testID='=' />
                </View>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    mainRow: {
        backgroundColor: 'darkgrey',
        flex: 1, 
    },
    container: {
      flex: 1,
      flexDirection: 'column',
      alignContent: 'space-evenly',
      justifyItems: 'space-evenly',
      justifyContent: 'flex-end',
      rowGap: 20,
    },
    button: {
        backgroundColor: 'cyan',
        borderRadius: 50,
        alignItems: 'center',
        justifyContent: 'center',
        width: 75,
        height: 75,
    },
    doubleWidthButton: { 
        width: 150, 
    },
    tripleWidthButton: {
        width: 225,
    },   
    text: {
        fontSize: 25,
    },
    row: {
        flex: 1,
        //backgroundColor: 'green',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        columnGap: 20,
    },
    sumArea: {
        flex: 1,
        backgroundColor: 'lightgray',
        alignItems: 'flex-end',
        justifyContent: 'center',
    },
    sum: {
        fontSize: 60,
    }
  });

export default CalculatorScreen;