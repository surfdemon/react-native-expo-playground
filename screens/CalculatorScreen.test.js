import React, { useState as useStateMock } from 'react';
import CalculatorScreen from './CalculatorScreen';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';

jest.setTimeout(7000);

describe('Calculator Screen Tests', () => {
    it('CalculatorScreen has one child', () => {
        const tree = renderer.create(<CalculatorScreen />).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it('Calculator Screen renders correctly', () => {
        const tree = renderer.create(<CalculatorScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Clear sums function should clear everything and set sum to 0', () => {
        const tree = renderer.create(<CalculatorScreen />).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it('sum should be 0 to start off with', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        const sumElement = getByTestId('sum');  
        console.log(sumElement.children[0]);  
        expect(sumElement.children[0]).toBe('0');
    });
    
    it('point press function should add a point to the sum to be 0. ', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('dot'));
        const sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('0.');
    });
    
    it('pressing .5, sum should be 0.5', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('dot'));
        fireEvent.press(getByTestId('5'));
        const sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('0.5');
    });

    it('pressing .5 +, sum should be 0', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('dot'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('+'));
        const sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('0');
    });

    it('pressing .5 + 5, sum should be 5', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('dot'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('+'));
        fireEvent.press(getByTestId('5'));
        const sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('5');
    });

    it('pressing 0.5 + 5 gives sum of 5.5', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('dot'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('+'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('='));
        sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('5.5');
    });

    it('pressing 5 + 5 gives sum of 10', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('+'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('='));
        sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('10');
    });

    it('pressing 0.5 * 5 gives sum of 2.5', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('dot'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('*'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('='));
        sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('2.5');
    });

    it('pressing 5 * 5 gives sum of 25', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('*'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('='));
        sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('25');
    });

    it('pressing 0.5 / 2 gives sum of 0.25', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('dot'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('/'));
        fireEvent.press(getByTestId('2'));
        fireEvent.press(getByTestId('='));
        sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('0.25');
    });

    it('pressing 10 / 2 gives sum of 5', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('1'));
        fireEvent.press(getByTestId('0'));
        fireEvent.press(getByTestId('/'));
        fireEvent.press(getByTestId('2'));
        fireEvent.press(getByTestId('='));
        sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('5');
    });

    it('pressing 10 - 2 gives sum of 5', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('1'));
        fireEvent.press(getByTestId('0'));
        fireEvent.press(getByTestId('-'));
        fireEvent.press(getByTestId('5'));
        fireEvent.press(getByTestId('='));
        sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('5');
    });

    it('currentValue should only allow one . to be entered', () => {
        const { getByTestId } = render(<CalculatorScreen />);
        fireEvent.press(getByTestId('dot'));
        fireEvent.press(getByTestId('dot'));
        const sumElement = getByTestId('sum');
        expect(sumElement.children[0]).toBe('0.');
    });
});