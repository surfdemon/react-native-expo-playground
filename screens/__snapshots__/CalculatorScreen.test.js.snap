// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Screen Tests Calculator Screen renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "darkgrey",
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignContent": "space-evenly",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "justifyItems": "space-evenly",
          "rowGap": 20,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "flex-end",
            "backgroundColor": "lightgray",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        id="sum"
        style={
          [
            {
              "fontSize": 25,
            },
            {
              "fontSize": 60,
            },
          ]
        }
        testID="sum"
      >
        0
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "columnGap": 20,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="C"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          C
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="*"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          *
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "columnGap": 20,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="1"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          1
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="2"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          2
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="3"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          3
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="+"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "columnGap": 20,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="4"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          4
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="5"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          5
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="6"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          6
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="-"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          -
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "columnGap": 20,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="7"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          7
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="8"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          8
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="9"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          9
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="/"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          /
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "columnGap": 20,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="0"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          0
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="dot"
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          .
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "cyan",
            "borderRadius": 50,
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="="
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          =
        </Text>
      </View>
    </View>
  </View>
</View>
`;
